feat(sdl): can set credentials

jzsfkzm opened the pull request:
One can select an image repository and enter credentials after this. Available in the SDL Builder, but not in the deployment UI.

ref #467

jzsfkzm committed:
feat(sdl): can set credentials

ref #467

jzsfkzm committed:
fix(sdl): auto-focus editor in import modal

baktun14 reviewed (commented):
The credentials for private repos are optional, since not all repos are private. The way you implemented it is great, we just need to add a flag [like "hasGpu"](https://github.com/akash-network/console/blob/main/apps/deploy-web/src/types/sdlBuilder.ts#L80) and conditionally add the validation. So a simple checkbox like the `hasGpu` to toggle the form controls should do it.

Is there a reason behind the `canChangeCredentials`? We would like to have it in all the SDL builder instances ([SDL Builder page](https://console.akash.network/sdl-builder) and [new deployment page](https://console.akash.network/new-deployment?step=edit-deployment)).

baktun14 commented on apps/deploy-web/src/utils/sdl/data.ts:
```diff
@@ -17,10 +17,17 @@ export const defaultHttpOptions = {
   id: nanoid(),
   title: "service-1",
   image: "",
+  credentials: defaultCredentials,
```
This is throwing and error in the build as the `username` and `password` are required.

jzsfkzm committed:
fix(sdl): credentials behind a checkbox, works for templates too

ref #467

jzsfkzm commented:
> The credentials for private repos are optional, since not all repos are private. The way you implemented it is great, we just need to add a flag [like "hasGpu"](https://github.com/akash-network/console/blob/main/apps/deploy-web/src/types/sdlBuilder.ts#L80) and conditionally add the validation. So a simple checkbox like the `hasGpu` to toggle the form controls should do it.
> 
> Is there a reason behind the `canChangeCredentials`? We would like to have it in all the SDL builder instances ([SDL Builder page](https://console.akash.network/sdl-builder) and [new deployment page](https://console.akash.network/new-deployment?step=edit-deployment)).

I added the flag `canChangeCredentials` because I figured it is not needed for templates on the Deploy page. Now I see it is very much needed there too, so removed. Now it is all the same for Deploy and SDL Builder.
Added a checkbox with label "Private" on top the box, revealing credential fields only when you click it.

baktun14 reviewed (approved):
I just tested it locally and it works great, great work!
